internal/action/infocomplete.go

```diff
@@ -81,9 +82,40 @@ func colorschemeComplete(input string) (string, []string) {
 func filetypeComplete(input string) (string, []string) {
 	var suggestions []string
 
-	for _, f := range config.ListRuntimeFiles(config.RTSyntax) {
-		if strings.HasPrefix(f.Name(), input) {
-			suggestions = append(suggestions, f.Name())
+	// Since we need to complete the file type instead of the file name
+	// we need to parse the RTSyntax* files and search for a matching FileType.
+	// The following logic can be simplified to the previos state again,
+	// in the moment the file names will represent the file types directly.
```

Comment:
The comment is confusing to a casual reader not familiar with the context (what "previous state"? "the file type instead of the file name" - wtf? which file name? etc).

Reply:
So I think the comment should be a simple clear explanation of the intention, something like: "We cannot match filetypes just by names of syntax files, since those names may be different from the actual filetype values specified inside syntax files (e.g. "c++" filetype in cpp.yaml). So we need to parse filetype values out of those files."